packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -485,6 +485,55 @@ export class Key extends KeyBase {
     return new Import(keyResourceName);
   }
 
+  /**
+   * Create a mutable {@link IKey} based on a low-level {@link CfnKey}.
+   * This is most useful when combined with the cloudformation-include module.
+   * This method is different than {@link fromKeyArn()} because the {@link IKey}
+   * returned from this method is mutable;
+   * meaning, calling any mutating methods on it,
+   * like {@link IKey.addToResourcePolicy()},
+   * will actually be reflected in the resulting template,
+   * as opposed to the object returned from {@link fromKeyArn()},
+   * on which calling those methods would have no effect.
+   */
+  public static fromCfnKey(cfnKey: CfnKey): IKey {
+    // use a "weird" id that has a higher chance of being unique
+    const id = '@FromCfnKey';
+
+    // if fromCfnKey() was already called on this cfnKey,
+    // return the same L2
+    // (as different L2s would conflict, because of the mutation of the keyPolicy property of the L1 below)
+    const existing = cfnKey.node.tryFindChild(id);
+    if (existing) {
+      return <IKey>existing;
+    }
+
+    let keyPolicy: iam.PolicyDocument;
+    try {
+      keyPolicy = iam.PolicyDocument.fromJson(cfnKey.keyPolicy);
+    } catch (e) {
```

Comment:
what's the downside to leaving the policy as `undefined` so users can still convert to an `IKey` that has an immutable policy? this limits the use but is similar to what we do for imported keys

Reply:
Because it would make the returned `IKey` immutable, which defeats the purpose of using `fromCfnKey()` in the first place.

Basically, if you set the policy as `undefined` in this `catch`, then `fromCfnKey()` would sometimes return a mutable `IKey`, and sometimes an immutable one, depending on what gets passed as the `CfnKey` parameter. That's a bad experience in my opinion - a customer would be very surprised if the permissions for a Key that got from `fromCfnKey()` were sometimes correctly updated, and sometimes not.

I think throwing, and making it explicit, is a much less error-prone experience, and also one we can make better in the future by throwing in less cases (see below).